Use events to refocus header in editor e2e for resiliency

dpasque opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

## Proposed Changes

I noticed in a test run recently that sometimes the block toolbar can cover the main header, which eats the click when we are trying to reset the block focus.

This makes that more resilient by just dispatching the click event directly.

## Testing Instructions

All builds should pass!

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Commit:
Use events to refocus header in editor e2e

This provides resiliency when a toolbar covers it.

dpasque commented:
Hmm, this doesn't seem to be working on CI as stably as I hoped, I need to do some more investigation

worldomonation commented on packages/calypso-e2e/src/lib/components/editor-gutenberg-component.ts:
```diff
@@ -45,7 +45,8 @@ export class EditorGutenbergComponent {
 		const locator = editorCanvas.locator( selectors.title );
-		await locator.click();
+		// Every now and then, a block toolbar can cover the title, so we "force" the click by doing it via the event directly.
+		await locator.dispatchEvent( 'click' );
```
What if, instead of selecting the `Title` block, we select the editor toolbar at the top?

This part:

![Screenshot 2023-10-13 at 4 06 28 PM](https://github.com/Automattic/wp-calypso/assets/6549265/dc4b7254-29d3-4d6e-9f87-a21f5f58d20d)

worldomonation replied:
Implemented and tried this method locally and it seems to work just fine.

I'd like to hear your thoughts on this; would "promoting" this method to the `EditorPage` be too much?

dpasque replied:
Ahhh, I think I finally figured it out!

I did some more testing, and unfortunately, the solution you had was also failing at the same points as mine was.

The reason for both... the underlying issue is just focus! What determines block availability is what currently has focus. Dispatching a `click` event doesn't actually set focus. (As opposed to manually clicking which both dispatched `click` and sets DOM focus).

So, I rolled back to my original solution and also added `locator.focus()`, and it works like a charm! 🎉 

Thank you so much for the extra help investigating! 🙇

worldomonation committed:
packages/calypso-e2e/src/lib/components/editor-gutenberg-component.ts
- remove `resetSelectedBlock` method.

packages/calypso-e2e/src/lib/pages/editor-page.ts
- add `resetSelectedBlock` method with a different implementation.

Commit:
Revert "packages/calypso-e2e/src/lib/components/editor-gutenberg-component.ts"

This reverts commit 18d10a3382f5c1e702d51c65029e82c4e60f9dc7.

Commit:
Also force focus to reset the block
